tests/test_user.py

```diff
@@ -111,6 +112,190 @@ def test_change_middle_name(self, driver, profile_settings_page, fake):
+        """
+        settings_page = user.AccountSettingsPage(driver)
+        settings_page.goto()
+        assert user.AccountSettingsPage(driver, verify=True)
+
+        # Enter an IMAP enabled email address in the email address input box and click
+        # the Add email button
+        settings_page.email_address_input.send_keys_deliberately(settings.IMAP_EMAIL)
+        settings_page.add_email_button.click()
+        try:
+            # After clicking the Close button on the confirmation modal, verify that the
+            # email address is displayed in the Unconfirmed emails list
+            settings_page.confrim_email_sent_modal.close_button.click()
+            unconfirmed_email = settings_page.get_unconfimred_email_item(
+                settings.IMAP_EMAIL
+            )
+            assert unconfirmed_email is not None
+
+            # Next connect to the email account and verify that a Confirm account merge
+            # email was actually sent
+            email_count = 0
+            loop_counter = 0
+            while email_count == 0:
+                # Retrieve count of UNSEEN email messages from inbox
+                email_count = EmailAccess.get_count_of_unseen_emails_by_imap(
+                    settings.IMAP_HOST,
+                    settings.IMAP_EMAIL,
+                    settings.IMAP_EMAIL_PASSWORD,
+                )
+                # To prevent an endless loop waiting for the email
+                loop_counter += 1
+                if loop_counter == 90:
+                    raise Exception(
+                        'No unseen emails. Verify that Reset Password email was sent.'
+                    )
+                    break
+
+            if email_count > 0:
+                # Next retrieve the email with subject 'Confirm account merge'
+                email_body = EmailAccess.get_latest_email_body_by_imap(
+                    settings.IMAP_HOST,
+                    settings.IMAP_EMAIL,
+                    settings.IMAP_EMAIL_PASSWORD,
+                    'Inbox',
+                    'SUBJECT',
+                    'Confirm account merge',
+                )
+                # Search through the email body and verify that the OSF account owner's
+                # email address is in the body of the email
+                assert str(email_body).find(settings.USER_ONE) > 0
+        finally:
+            # Lastly delete the unconfirmed email from the account
+            unconfirmed_email = settings_page.get_unconfimred_email_item(
+                settings.IMAP_EMAIL
+            )
+            if unconfirmed_email is not None:
+                delete_button = unconfirmed_email.find_element_by_css_selector(
+                    'button[data-test-delete-button]'
+                )
+                delete_button.click()
+                # Verify email address in text of confirmation modal
+                assert (
+                    settings_page.confrim_remove_email_modal.deleted_email.text
```

Comment:
Please confrim the spelling of "confrim" :)

Reply:
D'oh!  - Fixed.